File: data/src/main/java/org/oppia/data/backends/gae/model/CustomizationArgs.java
package org.oppia.data.backends.gae.model;

import com.squareup.moshi.Json;

public abstract class CustomizationArgs {

  /*
   * Parent Hierarchy: ExplorationContainer -> Exploration -> State -> ParamChange -> CustomizationArgs
   * Parent Hierarchy: ExplorationContainer -> Exploration -> ParamChange -> CustomizationArgs
   */

  @Json(name = "parse_with_jinja")
  public boolean parseWithJinja;

  //TODO: Check this free form data
  @Json(name = "value")
  public String value;

}

File: data/src/main/java/org/oppia/data/backends/gae/model/CustomizationArgs.kt
package org.oppia.data.backends.gae.model

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class CustomizationArgs (

  @Json(name="parse_with_jinja") val isParseWithJinja: Boolean?,
  @Json(name="value") val value: Any?

)
